docs_outdoornav_user_manual/web_user_interface/ui_autonomous_mode.mdx

```diff
@@ -213,6 +228,85 @@ To add a Task to the end of a Mission:
 
     :::
 
+### Custom Tasks  {#custom-task}
+
+Users can create their own tasks following a specific template and place them within the **/onav_log/custom_tasks** directory.
+Restarting the UGV or dockers will trigger the system to load the custom task that was created, making it available for mission use.
+Below is a sample template file named "log_inputs.py" which is elaborated further in the following section.
+
+#### Sample Custom Task
+
+```py
+#!/usr/bin/python3
+
+from onav_custom_tasks.custom_task_base import *
+
+class CustomTask(CustomTaskBase):
+  def __init__(self):
+      super().__init__("log_inputs")
+
+  def run_task(self, goal):
+    if len(goal.strings) == 0 or len(goal.floats) == 0:
+      self._as.set_aborted()
+      return False
+
+    for string in goal.strings:
+      self._feedback.state = string      
+      self._as.publish_feedback(self._feedback)
+      
+    for num in goal.floats:
+      self._feedback.state = str(num)
+      self._as.publish_feedback(self._feedback)
+
+    return True
+```
+
+This sample custom task iterates through the two available data variables and outputs them to the feedback topic. If neither of the 
+variables have any data in them the task is aborted.
+
+##### Custom Task Class
+
+```py 
+from onav_custom_tasks.custom_task_base import *
+
+class CustomTask(CustomTaskBase):
+  def __init__(self):
+      super().__init__("log_inputs")
+
+  def run_task(self, goal):
+```
+
+To leverage custom tasks users will need to create a new file that defines a new class called `CustomTask` which inherits from 
+`CustomTaskBase`. The `CustomTaskBase` class is found in the `onav_custom_tasks.custom_task_base` package. It exposes a `SimpleActionServer` 
+object as `_as`, a `UITaskFeedback` object as `_feedback` and a `UITaskResults` object as `_result` to be used as part of the tasks functionality. The custom 
+task manager requires the user to also define a `run_task` method that takes a `UITaskGoal` object as a parameter.
+
+##### Aborting Tasks 
+```py 
+if len(goal.strings) == 0 or len(goal.floats) == 0:
+  self._as.set_aborted()
+  return False
+```
+
+When running the task users may handle errors through the action servers `set_aborted()` function.  When this function is called the entire mission 
+will be aborted. In this sample the task is aborted if neither of the lists have any values.
+
+##### Setting feedback
+```py 
+for string in goal.strings:
+  self._feedback.state = string      
+  self._as.publish_feedback(self._feedback)
+  
+for num in goal.floats:
+  self._feedback.state = str(num)
+  self._as.publish_feedback(self._feedback)
+
+  return True
+```
+
+Users can set the feedback state using the `SimpleActionServer` class's standard methods. In this case the task iterates through the comma deliminated

```

Comment:
We should have a link to http://wiki.ros.org/actionlib_tutorials/Tutorials/Writing%20a%20Simple%20Action%20Client%20%28Python%29 or similar for the SimpleActionServer.